web/src/pages/api/public/score-configs.ts

```diff
@@ -1,109 +1,119 @@
-import { isPrismaException } from "@/src/utils/exceptions";
-import { isCastedConfig } from "@/src/features/manual-scoring/lib/helpers";
+import { type z } from "zod";
+import { Prisma, prisma } from "@langfuse/shared/src/db";
+import { isBooleanDataType } from "@/src/features/manual-scoring/lib/helpers";
+import { v4 } from "uuid";
+import { createAuthedAPIRoute } from "@/src/features/public-api/server/createAuthedAPIRoute";
+import { withMiddlewares } from "@/src/features/public-api/server/withMiddlewares";
+import {
+  PostScoreConfigResponse,
+  ScoreConfig,
+  GetScoreConfigsResponse,
+  GetScoreConfigsQuery,
+  PostScoreConfigBody,
+} from "@/src/features/public-api/types/score-configs";
+import * as Sentry from "@sentry/node";
+import { InvalidRequestError } from "@langfuse/shared";
 
-const ScoreConfigsGetSchema = z.object({
-  ...paginationZod,
-});
-
-export default async function handler(
-  req: NextApiRequest,
-  res: NextApiResponse,
-) {
-  await runMiddleware(req, res, cors);
+const inflateConfigBody = (body: z.infer<typeof PostScoreConfigBody>) => {
+  if (isBooleanDataType(body.dataType)) {
+    return {
+      ...body,
+      categories: [
+        { label: "True", value: 1 },
+        { label: "False", value: 0 },
+      ],
+    };
+  }
+  return body;
+};
 
-  try {
-    // CHECK AUTH
-    const authCheck = await verifyAuthHeaderAndReturnScope(
-      req.headers.authorization,
-    );
-    if (!authCheck.validKey) {
-      return res.status(401).json({
-        message: authCheck.error,
+export default withMiddlewares({
+  POST: createAuthedAPIRoute({
+    name: "Create Score Config",
+    bodySchema: PostScoreConfigBody,
+    responseSchema: PostScoreConfigResponse,
+    fn: async ({ body, auth }) => {
+      const existingConfig = await prisma.scoreConfig.findFirst({
+        where: {
+          projectId: auth.scope.projectId,
+          name: body.name,
+          dataType: body.dataType,
+        },
       });
-    }
-    // END CHECK AUTH
 
-    if (req.method !== "GET") {
-      return res.status(405).json({ message: "Method not allowed" });
-    }
-
-    if (req.method === "GET") {
-      if (authCheck.scope.accessLevel !== "all") {
-        return res.status(401).json({
-          message: "Access denied - need to use basic auth with secret key",
-        });
+      if (existingConfig) {
+        throw new InvalidRequestError(
+          "Score config with this name and data type already exists for this project",
+        );
       }
 
-      const obj = ScoreConfigsGetSchema.parse(req.query);
+      const inflatedConfigInput = inflateConfigBody(body);
+
+      const config = await prisma.scoreConfig.create({
+        data: {
+          ...inflatedConfigInput,
+          categories: inflatedConfigInput.categories ?? undefined,
+          id: v4(),
+          projectId: auth.scope.projectId,
+        },
+      });
 
+      return config as z.infer<typeof ScoreConfig>;
```

Comment:
the goal of this api wrapper is to throw type warnings in case the underlying db types drift from the api type. by defining the return type as this type, this type warning does not work anymore